packages/openneuro-server/src/graphql/resolvers/dataset.js

```diff
@@ -23,10 +24,22 @@ import { getDatasetWorker } from '../../libs/datalad-service.js'
 import { getDraftHead } from '../../datalad/dataset.js'
 import { getFileName } from '../../datalad/files.js'
 
-export const dataset = (obj, { id }, { user, userInfo }) => {
-  return checkDatasetRead(id, user, userInfo).then(() => {
-    return datalad.getDataset(id)
-  })
+export const dataset = async (
+  obj,
+  { id },
+  { user, userInfo },
+  ignoreInaccessibleDatasets = false,
+) => {
+  try {
+    return await checkDatasetRead(id, user, userInfo).then(() => {
+      return datalad.getDataset(id)
+    })
+  } catch (err) {
+    // Don't throw the error if this is for a dataset search.
+    if (ignoreInaccessibleDatasets && err.message === states.READ.errorMessage)
+      return null
+    else throw err
+  }
```

Comment:
I don't think the flag here makes sense, instead the error can be caught in the search resolver rather than overloading this one with search specific behavior. This also violates the signature for resolvers, which is `fieldName: (parent, args, context, info) => data`.